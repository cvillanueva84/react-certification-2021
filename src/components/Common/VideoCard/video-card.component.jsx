import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import formatter from '../../../utils/formatter';

const VideoCard = (props) => {
  const { snippet } = props;
  return (
    <Fragment>
      <Link
        to="/player"
        className="card video-card decoration-none"
        data-testid="video-card"
      >
        <img
          src={snippet.thumbnails.medium.url}
          className="card-img-top"
          alt="Video thumbnail"
        />
        <div className="card-body px-0">
          <h5 className="card-title">{snippet.title}</h5>
          <p className="m-0 card-lead">
            {snippet.channelTitle} &bull; {formatter.formatDate(snippet.publishTime)}
          </p>
        </div>
      </Link>
    </Fragment>
  );
};

export default VideoCard;
