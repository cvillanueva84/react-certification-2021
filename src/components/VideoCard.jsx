import React from 'react';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

export default function VideoCard({ videoInfo }) {
  const useIconStyles = makeStyles({
    cardImage: {
      width: '100%',
      height: '200px',
      backgroundImage: `url(${videoInfo.thumbnails.high.url})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      backgroundPosition: '50% 50%',
      '&:hover': {
        opacity: '0.8',
        cursor: 'pointer',
      },
    },
    title: {
      margin: '5px',
      fontSize: '14px',
    },
    description: {
      padding: '5px',
      fontSize: '10px',
    },
  });

  const classes = useIconStyles();

  return (
    <Paper
      elevation={3}
      style={{ height: '300px', width: '300px', margin: '20px', paddingBottom: '20px' }}
    >
      <Grid
        container
        direction="column"
        justifyContent="flex-start"
        alignItems="flex-start"
        style={{ padding: '0px' }}
      >
        <Grid item>
          <img
            src={videoInfo.thumbnails.high.url}
            className={classes.cardImage}
            alt={videoInfo.Title}
          />
        </Grid>
        <Grid item>
          <Typography variant="h6" className={classes.title}>
            {videoInfo.title}
          </Typography>
        </Grid>
        <Grid item>
          <Typography variant="h6" className={classes.description}>
            {videoInfo.description}
          </Typography>
        </Grid>
      </Grid>
    </Paper>
  );
}
