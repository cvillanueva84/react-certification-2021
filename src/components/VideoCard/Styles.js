import styled from 'styled-components';

export const Card = styled.div`
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  height: 330px;
  width: 330px;
  background-color: white;
  border-radius: 10px;
  border: 1px solid #ddd;
  margin: 10px;
  padding: 1em;
  cursor: pointer;

  img {
    width: 100%;
    height: 150px;
  }
  h5 {
    font-size: 28px;
    margin: 2px 5px 1px 5px;
    padding: 0px;
    font-family: Arial, Helvetica, sans-serif;
    @media (max-width: 1100px) {
      font-size: 2.2vw;
    }
    @media (max-width: 450px) {
      font-size: 12px;
    }
  }

  p {
    font-size: 14px;
    font-family: Arial, Helvetica, sans-serif;
    color: #a5a18f;
    margin: 0px 0px 5px 5px;
    padding: 15px;
    @media (max-width: 1100px) {
      font-size: 1.5vw;
    }
    @media (max-width: 450px) {
      font-size: 10px;
    }
  }
`;
