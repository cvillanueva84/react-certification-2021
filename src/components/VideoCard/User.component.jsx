import React from 'react';
import { UserContainer } from './User.styled';

export const User = () => {
  return (
    <UserContainer>
      <img src="https://via.placeholder.com/100x100" alt="userPicture" />
    </UserContainer>
  );
};

export default User;
