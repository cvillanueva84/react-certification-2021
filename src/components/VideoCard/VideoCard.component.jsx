import React, { useState, useRef, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { Paper } from '@material-ui/core';

import {
  VideoThumbnail,
  VideoInfo,
  VideoTitle,
  VideoDescription,
} from './VideoCard.module';

import './VideoCard.styles.css';

function HomeView(props) {
  const { title, thumbnail, description, videoId } = props;
  const [isCardHovered, setIsCardHovered] = useState(false);
  const cardRef = useRef(null);

  const handleMouseOver = () => {
    setIsCardHovered(true);
  };

  const handleMouseOut = () => {
    setIsCardHovered(false);
  };

  useEffect(() => {
    const node = cardRef.current;
    if (node) {
      node.addEventListener('mouseover', handleMouseOver);
      node.addEventListener('mouseout', handleMouseOut);
      return () => {
        node.removeEventListener('mouseover', handleMouseOver);
        node.removeEventListener('mouseout', handleMouseOut);
      };
    }
  });

  return (
    <Link to={`/video/${videoId}`}>
      <Paper className="video-card" elevation={3} ref={cardRef}>
        <VideoThumbnail>
          <img alt={title} src={thumbnail} />
        </VideoThumbnail>
        <VideoInfo>
          <VideoTitle>
            <span>{title}</span>
          </VideoTitle>
          <VideoDescription className={!isCardHovered && 'video-description-reduce'}>
            {description || 'No description :('}
          </VideoDescription>
        </VideoInfo>
      </Paper>
    </Link>
  );
}

export default HomeView;
