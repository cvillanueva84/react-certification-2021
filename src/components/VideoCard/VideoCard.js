import React from 'react';
import Grid from '@material-ui/core/Grid';

const VideoCard = ({ changeSelection, videoList, selectedVideoId }) => {

  const changeSelectedVideo = (data) => {
    changeSelection(data);
  };

  return (
    <>
      {videoList.length > 0 &&
        videoList.map((data) => {
          if (data?.id !== selectedVideoId) {
            return (
              <Grid
                item
                xs={8}
                sm={10}
                md={4}
                lg={3}
                xl={3}
                key={data?.id}
                onClick={() => {
                  changeSelectedVideo(data);
                }}
                style={{ cursor: 'pointer' }}
              >
                <div>
                  <div className="card">
                    <div className="card-cover-image">
                      <img src={data?.thumbnails.medium.url} alt="thumbnail" />
                    </div>
                    <div className="card-box">
                      <div className="card-title">
                        <span>{data?.title}</span>
                      </div>
                      <div className="card-subtitle">{data?.channel.title}</div>
                      <div className="card-content">
                        <p>{data?.description}</p>
                      </div>
                    </div>
                  </div>
                </div>
              </Grid>
            );
          }
        })}
    </>
  );
};

export default VideoCard;
