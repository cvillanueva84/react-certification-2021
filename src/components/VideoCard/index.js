export { default } from './Videocard.component';
