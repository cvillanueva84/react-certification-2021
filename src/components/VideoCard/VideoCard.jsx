import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from './Styles';
import { selectVideo } from '../../store/globalActions';
import { useGlobalProvider } from '../../store/global.provider';

export default function VideoCard({ data }) {
  const { title, description, url, videoId } = data;
  const { dispatch } = useGlobalProvider();

  const selectingVideo = () => {
    selectVideo(dispatch, data);
  };

  return (
    <>
      <Card data-testid="list-videos">
        <Link
          to={{
            pathname: `/video/${videoId}`,
          }}
          onClick={selectingVideo}
        >
          <img src={url} alt={title} />
          <h5>{title}</h5>
        </Link>
        <p>{description}</p>
      </Card>
    </>
  );
}
