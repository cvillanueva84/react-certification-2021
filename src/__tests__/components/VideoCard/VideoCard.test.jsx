import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import VideoCard from '../../../components/VideoCard/VideoCard.component';

let tree;
const thumbnail = 'img.jpg';
const title = 'title';
const description = 'description';
const videoId = '123456';

beforeEach(() => {
  tree = renderer
    .create(
      <BrowserRouter>
        <VideoCard
          thumbnail={thumbnail}
          title={title}
          description={description}
          videoId={videoId}
        />
      </BrowserRouter>
    )
    .toJSON();
});

describe('VideoCard', () => {
  it('Renders VideoCard', () => {
    expect(tree).toMatchSnapshot();
  });
});
