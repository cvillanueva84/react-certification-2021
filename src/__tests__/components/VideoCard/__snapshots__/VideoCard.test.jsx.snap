// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Renders VideoCard 1`] = `
<a
  href="/video/123456"
  onClick={[Function]}
>
  <div
    className="MuiPaper-root video-card MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      className="sc-bdnxRM Zckej"
    >
      <img
        alt="title"
        src="img.jpg"
      />
    </div>
    <div
      className="sc-gtsrHT ipAONl"
    >
      <div
        className="sc-dlnjwi xeLvv"
      >
        <span>
          title
        </span>
      </div>
      <div
        className="sc-hKFxyN qYhat video-description-reduce"
      >
        description
      </div>
    </div>
  </div>
</a>
`;
